clevertap-core/src/main/java/com/clevertap/android/sdk/DeviceInfo.java

```diff
@@ -619,6 +639,23 @@ public void incrementLocalInAppCount() {
         getDeviceCachedInfo().localInAppCount++;
     }
 
+    public String getDeviceLocale() {
+        return getDeviceCachedInfo().locale;
+    }
+
+    public void setCustomLocale(String customLocale) {
+        this.customLocale = customLocale;
+    }
+
+    public String getCustomLocale() {
+        return customLocale;
+    }
+
+    public String getLocale() {
+        // If locale is set by the client then use that, otherwise fetch it from the device
+        return TextUtils.isEmpty(getCustomLocale()) ? getDeviceLocale() : getCustomLocale();
```

Comment:
This method might return null. Please check

Reply:
If customLocale is null(not set by the user) then we use the deviceLocale which won't be null

Reply:
TextUtils.isEmpty() -> this one is also checking null internally, resolving, name is misleading.